plugins/inputs/cisco_telemetry_mdt_netconf/utils.go

```diff
@@ -0,0 +1,331 @@
+package cisco_telemetry_mdt_netconf
+
+import (
+	"encoding/xml"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+// State is a flag that tracks the state of an object
+type state uint
+
+// Types of state
+const (
+	initialized = iota + 1
+	updated
+)
+
+// Numerical base
+const base = 10
+
+type metricTracker struct {
+	testutil.Metric
+	measurementFlag  uint
+	measurementLevel uint
+}
+
+// lookupMeasurement composes the measurement and then searches for it
+// in a predefined list of xpaths
+func (m *metricTracker) lookupMeasurement(
+	xpaths map[string]interface{}, e stackElem) {
+	if m.measurementFlag == initialized {
+		if len(e.Tree.Children) > 0 {
+			if len(m.Measurement) != 0 {
+				m.Measurement = m.Measurement + "/"
+			}
+			m.Measurement = m.Measurement + e.Tree.XMLName.Local
+		}
+		if _, measurementFound := xpaths[m.Measurement]; measurementFound {
+			m.measurementFlag = updated
+		}
+		m.measurementLevel++
+	}
+}
```

Comment:
Why are you writing your own "XPath" parser? Honestly speaking it's not even xpath but just a _very_ small subset namely the _path_ part... So either use a real XPath parser as mentioned before or change your terminology (especially the user facing one) to not suggest someone can put a _real_ xpath expression here.